templates/apps/sumo-aws-apps/sumologic-app-utils/src/awsresource.py

```diff
@@ -721,8 +780,359 @@ def extract_params(self, event):
+                )
+
+                response = session_config.put_delivery_channel(
+                    DeliveryChannel={
+                        "name":"default",
+                        "s3BucketName": f"{ConfigBucket}",
+                        "configSnapshotDeliveryProperties": {
+                            "deliveryFrequency":f"{Frequency}"
+                        }
+                    }
+                )
+
+                response = session_config.start_configuration_recorder(
+                    ConfigurationRecorderName="default"
+                )
+
+                resource_groups_response = resource_groups_client.tag(
+                    Arn=f"{resource_group_arn}",
+                    Tags={
+                        'disable-awsconfig-when-remove-cfn':'true'
+                    }
+                )
+            else:
+                for config_recorder in config_recorder_response['ConfigurationRecordersStatus']:
+                    if not config_recorder['recording']:
+                        logger.info(f'account: {account_id} on region: {region} Config enabled, but not recording')
+                        channels = session_config.describe_delivery_channels()
+                        bucket_name=None
+                        if "DeliveryChannels" in channels:
+                            for channel in channels["DeliveryChannels"]:
+                                bucket_name = channel["s3BucketName"] if "s3BucketName" in channel else None
+                                break
+                        if bucket_name==None:
+                            response = session_config.put_delivery_channel(
+                                DeliveryChannel={
+                                    "name":"default",
+                                    "s3BucketName": f"{ConfigBucket}",
+                                    "configSnapshotDeliveryProperties": {
+                                        "deliveryFrequency":f"{Frequency}"
+                                    }
+                                }
+                            )
+                            resource_groups_response = resource_groups_client.tag(
+                                Arn=f"{resource_group_arn}",
+                                Tags={
+                                    'remove-s3-in-delivery-channel-when-remove-cfn':'true'
+                                }
+                            )                        
+                        resource_groups_response = resource_groups_client.tag(
+                            Arn=f"{resource_group_arn}",
+                            Tags={
+                                'stop-delivery-channel-when-remove-cfn':'true'
+                            }
+                        )
+                        response = session_config.start_configuration_recorder(
+                            ConfigurationRecorderName="default"
```

Comment:
Can we create only one recorder in a region ?  What if the recorder with default name already exists?

Reply:
if recorder with default name already exists. i will check RecordersStatus. if not recording, it will start recorder, else recording, it will bypass